src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -93,7 +92,47 @@ def extract_dataset(args):
     files_to_check = []
     file_already_exist = []
 
+    def _check_system_python_version():
+        python_version_major = sys.version_info.major
+        python_version_minor = sys.version_info.minor
+        python_version_micro = sys.version_info.micro
+
+        # Set python version range
+        # python_range = 0 for python_version>="3.7"
+        # python_range = 1 for python_version>="3.5.3" and python_version<"3.7"
+        # python_range = 2 for python_version<"3.5.3"
+        if python_version_major == 3 and python_version_minor >= 7:
+            python_range = '0'
+        elif python_version_major == 3 and ((python_version_minor > 6) or (python_version_minor == 5 and python_version_micro >= 3)):
+            python_range = '1'
+        else:
+            python_range = '2'
+        return python_range
+
+    def _check_pandas_version(python_range, current_pandas_version, pandas_version_range):
+        # Valid pandas versions based on python versions
+        # python_range = 0; pandas==1.3.5
+        # python_range = 1; pandas>=0.23.3,<=0.25.3
+        # python_range = 2; pandas>=0.23.3,< 0.25.0
+        pd_0, pd_1, pd_2 = map(int,current_pandas_version.split("."))
+        if (python_range == '0' and current_pandas_version == '1.3.5') or (pd_0 == 0 and ((pd_1 == 23 and pd_2 >= 3) or (pd_1 == 24) or (pd_1 == 25 and pd_2 < 3) or (pd_1 == 25 and pd_2 == 3 and python_range==1))):
```

Comment:
Minor: a few ideas if you'd like to make the version comparisons slightly more readable:

1. Utilize the `Version` (part of `setuptools` so almost certainly already available but not necessarily - unless we declare this dependency already, didn't check):
```
import platform
from packaging.version import Version
python = Version(platform.python_version())
python_min = Version('3.5.3')
if (python >= python_min ...)
```
2. Make the versions tuples for easy compare:
```
import sys
python = sys.version_info[:3]
python_min = (3,5,3)
if (sys.version_info >= python_min ...)
```
In either way I would recommend formalizing what you already have in comment (lines 114-116) - list with valid version range combinations, for example:
```
valid_version_combinations = [ # ( python_min, python_max, pandas_min, pandas_max ), ...
( Version('0.0.0'), Version('3.5.3'), Version('0.23.3'), Version('0.25.0') ),
( Version('3.5.3'), Version('3.7.0'), Version('0.23.3'), Version('0.25.3') ),
( Version('3.5.7'), Version('999'), Version('1.3.5'), Version('1.3.5') ),
]
python = Version(platform.python_version())
pandas = Version(current_pandas_version)
# for simple positive test use `any(...)`, for inverse `not all(...)`
if ( any([python_min <= python and python <= python_max and pandas_min <= pandas and pandas <= pandas_max for python_min, python_max, pandas_min, pandas_max in valid_version_combinations]) ):
```

Reply:
1- We would need to `pip install packaging` to use Versions. Trying to avoid any further package installs in dx-toolkit.
2- Good suggestion. Will try using tuple to make the code more cleaner. Thanks!

Reply:
Updated! Thanks!